lib/std/fmt.zig

```diff
@@ -367,11 +370,26 @@ pub const Parser = struct {
     }
 
     // Returns the n-th next character or null if that's past the end
-    pub fn peek(self: *@This(), n: usize) ?u8 {
-        return if (self.pos + n < self.buf.len) self.buf[self.pos + n] else null;
+    pub fn peek(self: *@This(), n: usize) ?u21 {
+        const original_i = self.iter.i;
+        defer self.iter.i = original_i;
+
+        var i = 0;
+        var code_point: ?u21 = null;
+        while (i <= n) : (i += 1) {
+            code_point = self.iter.nextCodepoint();
+            if (code_point == null) return null;
+        }
+        return code_point;
     }
 };
 
+fn utf(char: u21) ![]const u8 {
+    var bytes: [4]u8 = undefined;
+    const len: u3 = try std.unicode.utf8Encode(char, &bytes);
+    return bytes[0..len];
+}
+
```

Comment:
This function returns a pointer to a stack variable, which is no longer valid after the function returns, see below for an alternative.
```suggestion
```

Reply:
> Also, some of the `@compileError`s in `fmt.Placeholder.parse` now trigger other compile errors before the custom compile error is produced. Something like the following function added to `unicode.zig` could be useful for fixing these:

good idea, actually the utf fn was comptime before but `options.fill` is only known at runtime. But this brings me now to the idea to replace it with the whole fill sequence which is indeed known at comptime.